.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,222 @@
+This document defines many guidelines that should be adhered to when developing 
+against Banshee. These guidelines will make the codebase more readable,
+extensible, and portable.
+
+Commit Message Guidelines
+=========================
+
+Every change to source code must have a commit message associated to it. The
+formatting details of this message are described here:
+
+  http://live.gnome.org/Git/CommitMessages
+
+Please review these guidelines separate to this document, and take in account
+these policies on top:
+
+  1. Unlike many other open-source projects, we don't use the git tagging
+     'Signed-off-by' to refer as a licencing statement of a contribution,
+     rather we use it as 'Reviewed by this maintainer' metadata.
+
+  2. As a consequence of the above, the choice of licencing attribution
+     is implicit: by contributing a patch to the Banshee project you are
+     automatically accepting to follow the same licence as the project
+     itself (see our COPYING file) unless stated on the contrary.
+
+  3. When a contributed patch contains new files, please include licence
+     headers in each of them, in the same way the rest of the files of
+     the projects already have.
+
+  4. It's recommended that you add yourself (or your company) to the list
+     of the copyright owners of a file if you're modifying a considerable
+     proportion of it.
+
+
+C# Coding Style Guidelines
+==========================
+
+These guidelines should be followed when writing code in Banshee. For the most
+part they are similar to the Mono syntax guidelines [1]. All public API must
+adhere to the .NET Framework Design Guidelines. [2]
+
+Patches and additions to the code base will be checked for adherence to these
+guidelines. If code is in violation, you will be asked to reformat it.
+
+  1. Private variable/field names should be written like:
+
+      lower_case_with_under_scores
```

Comment:
I'd really like to see this one changed. Not sure if it's feasible.

`testVar` just looks more natural to me in C#. Python would be something else...